src/Common/src/CoreLib/System/Globalization/DateTimeParse.cs

```diff
@@ -4511,8 +4534,9 @@ internal static bool TryParseQuoteString(ReadOnlySpan<char> format, int pos, Str
             bool bTimeOnly = false;
             result.calendar = parseInfo.calendar;
 
-            if (parseInfo.calendar.ID == CalendarId.HEBREW)
+            if (!GlobalizationMode.Invariant && (CalendarId)parseInfo.calendar.ID == CalendarId.HEBREW)
             {
+                LazyInitializer.EnsureInitialized (ref m_hebrewNumberParser, () => new MatchNumberDelegate(DateTimeParse.MatchHebrewDigits));
                 parseInfo.parseNumberDelegate = m_hebrewNumberParser;
```

Comment:
We should not statically initialize this via `.cctor` when we don't need it (like for instance in invariant mode).